### ISSUE TITLE
applicationinsights-agent-3.0.0-PREVIEW.2 with Spring configuration

### ISSUE THREAD

**Issue by LennartHaggkvistHM** [microsoft/ApplicationInsights-Java#1230]
### Expected behavior
Start as usual using the old (and documented) configuration

### Actual behavior
Exception:
```
2020-04-15 11:31:49.005Z ERROR i.opentelemetry.auto.bootstrap.Agent - Throwable thrown while installing the agent
java.lang.reflect.InvocationTargetException: null
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:564)
        at io.opentelemetry.auto.bootstrap.Agent.startAgent(Agent.java:181)
        at io.opentelemetry.auto.bootstrap.Agent.start(Agent.java:75)
        at com.microsoft.applicationinsights.agent.bootstrap.MainEntryPoint.start(MainEntryPoint.java:62)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:564)
        at io.opentelemetry.auto.bootstrap.AgentBootstrap.agentmain(AgentBootstrap.java:77)
        at io.opentelemetry.auto.bootstrap.AgentBootstrap.premain(AgentBootstrap.java:67)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:564)
        at java.instrument/sun.instrument.InstrumentationImpl.loadClassAndStartAgent(InstrumentationImpl.java:513)
        at java.instrument/sun.instrument.InstrumentationImpl.loadClassAndCallPremain(InstrumentationImpl.java:525)
Caused by: java.lang.reflect.InvocationTargetException: null
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:564)
        at io.opentelemetry.auto.tooling.AgentInstaller.installBytebuddyAgent(AgentInstaller.java:84)
        ... 19 common frames omitted
Caused by: com.microsoft.applicationinsights.agent.bootstrap.configuration.ConfigurationBuilder$ConfigurationException: No connection string or instrumentation key provided
        at com.microsoft.applicationinsights.agent.internal.BeforeAgentInstaller.start(BeforeAgentInstaller.java:105)
        at com.microsoft.applicationinsights.agent.internal.BeforeAgentInstaller.beforeInstallBytebuddyAgent(BeforeAgentInstaller.java:81)
        at io.opentelemetry.auto.tooling.BeforeAgentInstaller.beforeInstallBytebuddyAgent(BeforeAgentInstaller.java:9)
        ... 24 common frames omitted
2020-04-15 11:31:49.006Z ERROR c.m.applicationinsights.agent - Agent failed to start.
java.lang.IllegalStateException: Agent should have been started already
        at io.opentelemetry.auto.bootstrap.Agent.installAgentTracer(Agent.java:191)
        at io.opentelemetry.auto.bootstrap.Agent.start(Agent.java:102)
        at com.microsoft.applicationinsights.agent.bootstrap.MainEntryPoint.start(MainEntryPoint.java:62)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:564)
        at io.opentelemetry.auto.bootstrap.AgentBootstrap.agentmain(AgentBootstrap.java:77)
        at io.opentelemetry.auto.bootstrap.AgentBootstrap.premain(AgentBootstrap.java:67)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:564)
        at java.instrument/sun.instrument.InstrumentationImpl.loadClassAndStartAgent(InstrumentationImpl.java:513)
        at java.instrument/sun.instrument.InstrumentationImpl.loadClassAndCallPremain(InstrumentationImpl.java:525)
```

### To Reproduce
Take a working Spring Boot application that is using the 2.x version of the agent.
We used applicationinsights-agent-2.6.0-BETA.2.jar
together with 
```
<dependency>
            <groupId>com.microsoft.azure</groupId>
            <artifactId>applicationinsights-web-auto</artifactId>
            <version>2.6.0-BETA.3</version>
        </dependency>
        <dependency>
            <groupId>com.microsoft.azure</groupId>
            <artifactId>applicationinsights-core</artifactId>
            <version>2.6.0-BETA.3</version>
        </dependency>
        <dependency>
            <groupId>com.microsoft.azure</groupId>
            <artifactId>applicationinsights-web</artifactId>
            <version>2.6.0-BETA.3</version>
        </dependency>
```

The issue is probably that the old version worked when using the spring configuration (application.yaml):

```
azure:
  application-insights:
    instrumentation-key: VALUE_TO_BE_OVERRIDDEN
```

The new one seems to require hard coded instrumentation keys in the XML file. We don't want that since we are overriding this value from the release pipeline (we are using containers/AKS)

### System information
Please provide the following information:
 - SDK Version: 2.6.0-BETA.3
 - OS type and version:
 - Application Server type and version (if applicable): Spring boot
 - Using spring-boot? yes
 - Additional relevant libraries (with version, if applicable): applicationinsights-agent-3.0.0-PREVIEW.2.jar
 
### Logs
[Turn on SDK logs](https://docs.microsoft.com/en-us/azure/application-insights/app-insights-java-troubleshoot#debug-data-from-the-sdk) and attach/paste them to the issue. If using an application server, also attach any relevant server logs.

_Be sure to remove any private information from the logs before posting!_

**Comment by LennartHaggkvistHM** [comment]
Tried to upgrade SDK to 2.6.1-BETA but no difference.
```
<dependency>
            <groupId>com.microsoft.azure</groupId>
            <artifactId>applicationinsights-spring-boot-starter</artifactId>
            <version>2.6.1-BETA</version>
        </dependency>
        <dependency>
            <groupId>com.microsoft.azure</groupId>
            <artifactId>applicationinsights-web-auto</artifactId>
            <version>2.6.1-BETA</version>
        </dependency>
        <dependency>
            <groupId>com.microsoft.azure</groupId>
            <artifactId>applicationinsights-core</artifactId>
            <version>2.6.1-BETA</version>
        </dependency>
        <dependency>
            <groupId>com.microsoft.azure</groupId>
            <artifactId>applicationinsights-web</artifactId>
            <version>2.6.1-BETA</version>
        </dependency>
```

**Comment by trask** [comment]
Hi @LennartHaggkvistHM, for 3.0-Preview, you can specify the connection string using environment variable also: https://github.com/microsoft/ApplicationInsights-Java/wiki/3.0-Preview:-Configuration-Options#connection-string

**Comment by LennartHaggkvistHM** [comment]
Thanks, I did miss that part of the documentation. It would be good to include that in how to upgrade from 2.x (for example https://docs.microsoft.com/en-us/azure/azure-monitor/app/java-in-process-agent#upgrading-from-application-insights-java-sdk-2x)
(also an update of the spring-boot starter would be good)

In in unrelated thing: on https://github.com/microsoft/ApplicationInsights-Java/wiki/3.0-Preview:-Configuration-Options#application-log-capture  
there is the example:
```
 "threshold": "error"
```
So I thought I would be able to write "debug" - but nothing happened. When I changed to "DEBUG" it started to work, perhaps good to update the example?
